gto/api.py

```diff
@@ -424,42 +418,41 @@ def _show_versions(  # pylint: disable=too-many-locals
 ):
     """List versions of artifact"""
 
-    if isinstance(repo, str):
-        repo = Repo(repo)
-
     def format_hexsha(hexsha):
         return hexsha[:7] if truncate_hexsha and is_hexsha(hexsha) else hexsha
 
     shortcut = parse_shortcut(name)
     if shortcut.shortcut and ref:
         raise WrongArgs("Cannot specify both shortcut and ref")
 
-    with GitRegistry.from_repo(repo=repo) as reg:
+    with GitRegistry.from_url(repo) as reg:
         if raw:
             return reg.find_artifact(shortcut.name).versions
 
         artifact = reg.find_artifact(shortcut.name)
 
-    versions = _get_versions(
-        artifact,
-        deprecated,
-        registered_only,
-        assignments_per_version,
-        versions_per_stage,
-        sort,
-    )
-    if ref:
-        ref_hexsha = resolve_ref(repo, ref).hexsha
-        versions = [v for v in versions if v["commit_hexsha"] == ref_hexsha]
-
-    elif shortcut.latest:
-        versions = versions[:1]
-    elif shortcut.version:
-        versions = [v for v in versions if shortcut.version == v["version"]]
-    elif shortcut.stage:
-        versions = [
-            v for v in versions for a in v["stages"] if shortcut.stage == a["stage"]
-        ]
+        versions = _get_versions(
+            artifact,
+            deprecated,
+            registered_only,
+            assignments_per_version,
+            versions_per_stage,
+            sort,
+        )
+        if ref:
+            try:
+                ref_hexsha = reg.scm.resolve_rev(ref)
+            except RevError as e:
+                raise RefNotFound(ref=ref) from e
```

Comment:
just a comment - now the rev vs ref terminology disparity is quite funky

Reply:
agreed, see my other comment on ref vs rev, we can follow up on refactoring this some more after this PR